jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/TaskInstanceServiceImpl.java

```diff
@@ -143,12 +145,66 @@ public long addTask(Task task, ContentData contentData) {
             persistenceContext.persistContent(content);
             persistenceContext.setDocumentToTask(content, contentData, task);
         }
-        
-        
-        taskEventSupport.fireAfterTaskAdded(task, context);
+        fireEvents(task);
         return task.getId();
     }
 
+    private void fireEvents(Task task) {
+        fireWithStatus(task, Status.Created, this::fireAfterTaskAdded);
+        if (task.getTaskData().getStatus() != Status.Created) {
+            fireWithStatus(task, Status.Created, this::fireBeforeTaskActivated);
+            taskEventSupport.fireAfterTaskActivated(task, context);
+        }
+    }
+
+    private void fireWithStatus(Task task, Status eventStatus, Consumer<Task> consumer) {
```

Comment:
it's only invoked with `Status.Created ` so this param could be taken out from the input params of the method:
```suggestion
    private void fireWithStatus(Task task, Consumer<Task> consumer) {
```

Reply:
This was on purpose in case in fuure we need to publish an event with different status in this class